### ISSUE TITLE
Fix confusing wording about "test suite", "test", and "test case"

### ISSUE THREAD

**Issue by yashi** [zephyrproject-rtos/zephyr#74148]
**Describe the bug**

The first sub-section in the "[Test Framework](https://docs.zephyrproject.org/latest/develop/test/ztest.html#test-framework)" is "[Creating a test suite](https://docs.zephyrproject.org/latest/develop/test/ztest.html#creating-a-test-suite)".  The sub-section states:

> Using Ztest to create a test suite is as easy as calling the `ZTEST_SUITE`.

Thus, reader, including me, assumes that a "test suite" is something we define using `ZTEST_SUITE()`.  Thest first argument to the macro is `SUITE_NAME` as documented in https://docs.zephyrproject.org/apidoc/latest/ztest__test_8h.html .

The next sub-section is "[Adding tests to a suite](https://docs.zephyrproject.org/latest/develop/test/ztest.html#adding-tests-to-a-suite)". It says:

> [ZTEST](https://docs.zephyrproject.org/latest/develop/test/ztest.html#c.ZTEST) `(suite_name, test_name)` - Which can be used to add a test by `test_name` to a given suite by `suite_name`.

So if you have

```c
ZTEST(my_suite, test_my_test)
{
}
```

We all assume `my_suite` is a test suite and `test_my_test` is a test.  In fact, if you ask twister to print list of tests, it prints the test I just added:

```
$ ./scripts/twister ... --list-tests
ZEPHYR_BASE unset, using "/home/yashi/src/libcsp-zephyr/zephyr"
Install pyserial python module with pip to use --device-testing option.
Renaming output directory to /home/yashi/src/libcsp-zephyr/zephyr/twister-out.2
INFO    - Using Ninja..
INFO    - Zephyr version: v3.6.0-5868-g0a3f2f0397a8
INFO    - Using 'zephyr' toolchain.
 - sec.sub.my_test
```

So fa so good.

Now if you go to the next section, [Test Runner (Twister)](https://docs.zephyrproject.org/latest/develop/test/twister.html#test-runner-twister), 

**Expected behavior**

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

**Logs and console output**
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

**Comment by golowanow** [comment]
closed as incomplete duplicate of #74151

**Comment by yashi** [comment]
Thanks!
